### PR TITLE
pipe buffer size for log.pipe should be configurable

### PR DESCRIPTION
**Pull request by ericcarlschwartz**
Conflicts:
	proxy/logging/YamlLogConfig.cc

^^^ 
cherry-picked and resolved

### CODE REVIEW

**Comment by ericcarlschwartz** [comment]
ah shoot, I tested this on RHEL7 but looks like there might be some issues on the older freeBSD version with some of the fcntl stuff. advice on what to do about that would be appreciated! could do #ifdef guarding or something?

File: proxy/logging/LogFile.cc [hunk 1]
```diff
@@ -185,6 +187,32 @@ LogFile::open_file()
       Debug("log-file", "no readers for pipe %s", m_name);
       return LOG_FILE_NO_PIPE_READERS;
     }
+
+#ifdef __FreeBSD__
+    // we don't do this for FreeBSD
+#else
```
Comment (masaori335) [inline proxy/logging/LogFile.cc]: Please add a #ifdef for Linux :(

Comment (maskit) [reply proxy/logging/LogFile.cc]: or check if F_GETPIPE_SZ is available (defined).

**Comment by maskit** [comment]
The two commits should have been squashed, right? "address review comments" is obviously not a commit we want to see separately.

File: proxy/logging/LogFile.cc (hunk 1 above)
Comment (masaori335) [reply proxy/logging/LogFile.cc]: it's better.

Comment (ericcarlschwartz) [reply proxy/logging/LogFile.cc]: ah really sorry about this one i didn't realize it had been merged. confused why the PR build passed but the other didn't?

**Comment by ericcarlschwartz** [comment]
they should have, sorry

**Comment by maskit** [comment]
Never mind, Eric. Committers are responsible for it.

**Comment by zwoop** [comment]
Cherry-picked to v9.0.x branch.

@ericcarlschwartz @shinrich Please, going forward, be careful with the multiple commits (as been pointed out, it should have been squashed). Also, the "merge conflict" messages is alarming, but I hope just a mistake. There should never be merge conflicts on commits going into master, and therefore, you should never have that in a "master" commit message (ever).

**Pull request merged**